ISPy/util/neural_estimation.py

```diff
@@ -0,0 +1,631 @@
+import copy
+import numpy as np
+import os
+import random
+import sys
+import time
+import warnings
+from contextlib import redirect_stdout
+import ipdb
+
+with warnings.catch_warnings():
+    warnings.filterwarnings("ignore", category=FutureWarning)
+# To deactivate warnings: https://github.com/tensorflow/tensorflow/issues/7778
+os.environ['TF_CPP_MIN_LOG_LEVEL'] = '2'
+
+try:
+    import keras.backend.tensorflow_backend as ktf
+    from keras.callbacks import Callback, ModelCheckpoint
+    from keras.layers import Input, Conv2D, add, Activation
+    from keras.layers import Lambda
+    from keras.layers.advanced_activations import ELU
+    from keras.layers.merge import concatenate
+    from keras.models import Model
+    from keras.optimizers import Adam
+    from keras.regularizers import l2
+    from keras.utils.data_utils import Sequence
+    # sess = tf.Session(config=tf.ConfigProto(log_device_placement=True,allow_soft_placement=True))
+
+except ImportError:
+    print('Please install keras and tensorflow to continue.')
+    sys.exit()
+
+try:
+    import sparsetools as sp
+except ImportError:
+    print('Please install sparsetools to continue.')
+    sys.exit()
+
+
+# ==================================================================================
+def rotate(cubo, angle_index):
+    # Rotate cubes a multiple of 90 degrees.
+    nnqq = cubo.shape[2]
+    ncubo = np.copy(cubo)
+    for ii in range(nnqq):
+        ncubo[:, :, ii] = np.rot90(cubo[:, :, ii], angle_index)
+    return ncubo
```

Comment:
Useful, but maybe better placed in some `imtools.py` (or something similar)? Also, `cubo` (check for those in the remainder). ;)

Reply:
Moved to gentools (from generic tools)